pkg/logs/logs.go

```diff
@@ -50,12 +51,21 @@ func Start() error {
 	adScheduler = scheduler.NewScheduler(sources, services)
 
 	// setup the server config
-	endpoints, err := config.BuildEndpoints()
+	endpoints, err := config.BuildHTTPEndpoints()
```

Comment:
looks weird to define endpoints twice, I would either change the name to httpEndpoints or make it visible only in the http check context, eg:
```
...
// check if the agent can connect to the HTTP endpoint
if endpoints, err := config.BuildHTTPEndpoints(); err == nil {
  httpConnectivity = http.CheckConnectivity(endpoints.Main)
}
endpoints, err := config.BuildEndpoints(httpConnectivity)
...
```